enacts/calc.py

```diff
@@ -345,39 +382,72 @@ def cess_date(
 
     Parameters
     ----------
-    soil_moisture : DataArray
-        Array of daily soil moisture.
+    daily_data : DataArray
+        Array of daily soil moisture or daily rainfall.
     dry_thresh : float
-        Soil moisture threshold to determine dry day if `dry_thresh` is less than `soil_moisture`.
-    min_dry_days : int
+        Soil moisture threshold to determine a dry day.
+    dry_spell_length : int
         Minimum number of dry days in a row to be considered a dry spell.
-    time_coord : str, optional
-        Time coordinate in `soil_moisture` (default `time_coord`="T"). 
+    et : DataArray, optional
+        Evapotranspiration used by `water_balance` if `daily_data` is rainfall.
+        Can be a single value with no dimensions or axes.
+    taw : DataArray, optional
+        Total available water used by `water_balance` if `daily_data` is rainfall.
+        Can be a single value with no dimensions or axes.
+    sminit : DataArray, optional
+        Soil moisture initialization used by `water_balance` if `daily_data` is rainfall.
+        If DataArray, must not have `time_dim` dim.
+        Can be a single value with no dimensions or axes.
+    time_dim : str, optional
+        Time coordinate in `soil_moisture` (default `time_dim`="T"). 
+    
     Returns
     -------
     cess_delta : DataArray[np.timedelta64] 
-        Difference between first day of `soil_moisture` 
+        Difference between first day of `daily_data` 
         and cessation date.
+    
     See Also
     --------
+    water_balance
+
     Notes
     -----
     Examples
     --------
     """
-    dry_day = soil_moisture < dry_thresh
-    dry_spell = dry_day * 1
-    dry_spell_roll = dry_spell.rolling(**{time_coord: min_dry_days}).sum() == min_dry_days
-    cess_mask = dry_spell_roll * 1
-    cess_mask = cess_mask.where((cess_mask == 1))
-    cess_delta = cess_mask.idxmax(dim=time_coord)
+    # Initializing
+    spell_length = ((daily_data < dry_thresh)
+                    .rolling(**{time_dim: dry_spell_length})
+                    .sum()
+                    .dropna(time_dim)
+                    .isel({time_dim: 0})
```

Comment:
It was laziness to get the right T associated with the result. It turns out it was wrong because just doing .sum() ignores whether there was contiguous dry days, and whether there was an interruption with a wet day.

So now it just initializes with day 1. Per definition, no cess date will be found over the first few iterations since there is a minimum length threshold to reach.

And all that is likely to change when covering the case where input is rain and the water balance must be done first before assessing spells, and thus further integrated in the loop.